Lantiq Image: allow definitions of profiles with sub-profiles
- the list of sub-profiles is specified in the form "SUBPROFILES_$(PROFILE)"

(some templates need a # as first line to be called by "foreach" because of a gmake bug)

diff --git a/target/linux/lantiq/image/Makefile b/target/linux/lantiq/image/Makefile
--- a/target/linux/lantiq/image/Makefile
+++ b/target/linux/lantiq/image/Makefile
@@ -83,6 +83,7 @@ define MkImageGzip
 endef
 
 define Image/Build/squashfs
+	# FIXME: leave this line to allow foreach loops
 	cat $(KDIR)/uImage-$(2) $(KDIR)/root.$(1) > $(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(1).image
 	$(call prepare_generic_squashfs,$(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(1).image)
 	$(if $(3),$(call MkBrnImage,$(3),$(4),$(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(3)-brnImage,$(2),$(1),$(5)))
@@ -171,29 +172,32 @@ define Image/BuildEVA/jffs2-128k
 endef
 
 define Image/Build/jffs2-64k
+	# FIXME: leave this line to allow foreach loops
 	dd if=$(KDIR)/uImage-$(2) of=$(KDIR)/uImage-$(2)-$(1) bs=64k conv=sync
 	cat $(KDIR)/uImage-$(2)-$(1) $(KDIR)/root.$(1) > $(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(1).image
 endef
 
 define Image/Build/jffs2-128k
+	# FIXME: leave this line to allow foreach loops
 	dd if=$(KDIR)/uImage-$(2) of=$(KDIR)/uImage-$(2)-$(1) bs=128k conv=sync
 	cat $(KDIR)/uImage-$(2)-$(1) $(KDIR)/root.$(1) > $(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(1).image
 endef
 
 define Image/Build/jffs2-256k
+	# FIXME: leave this line to allow foreach loops
 	dd if=$(KDIR)/uImage-$(2) of=$(KDIR)/uImage-$(2)-$(1) bs=256k conv=sync
 	cat $(KDIR)/uImage-$(2)-$(1) $(KDIR)/root.$(1) > $(BIN_DIR)/$(IMG_PREFIX)-$(2)-$(1).image
 endef
 
 define Image/Build/ubifs
-
+	# FIXME: leave this line to allow foreach loops
 ifneq ($($(PROFILE)_UBIFS_OPTS),)
 	$(CP) $(KDIR)/root.ubifs $(BIN_DIR)/$(IMG_PREFIX)-$(PROFILE)-rootfs.ubifs
 endif
 endef
 
 define Image/Build/ubi
-
+	# FIXME: leave this line to allow foreach loops
 ifneq ($($(PROFILE)_UBI_OPTS),)
 	$(CP) $(KDIR)/root.ubi $(BIN_DIR)/$(IMG_PREFIX)-$(PROFILE)-rootfs.ubi
 	$(CP) $(KDIR)/root-overlay.ubi $(BIN_DIR)/$(IMG_PREFIX)-$(PROFILE)-rootfs-overlay.ubi
@@ -213,7 +217,7 @@ endef
 
 
 define Image/InstallKernel/Template
-
+	# FIXME: leave this line to allow foreach loops
 ifneq ($(CONFIG_TARGET_ROOTFS_INCLUDE_KERNEL),)
 	$(INSTALL_DIR) $(TARGET_DIR)/boot
 
@@ -238,6 +242,7 @@ endif
 endef
 
 define Image/BuildKernel/Template
+	# FIXME: leave this line to allow foreach loops
 	$(call PatchKernelLzma,$(1),,$(if $(DTS_$(1)),$(DTS_$(1))))
 	$(call MkImageLzma,$(1))
 	$(CP) $(KDIR)/uImage-$(1) $(BIN_DIR)/$(IMG_PREFIX)-$(1)-uImage
@@ -445,15 +450,19 @@ endif
 
 
 define Image/BuildKernel
-	$(call Image/BuildKernel/Profile/$(PROFILE))
+	$(foreach profile,$(if $(SUBPROFILES_$(PROFILE)),$(SUBPROFILES_$(PROFILE)),$(PROFILE)),\
+		$(call Image/BuildKernel/Profile/$(profile)) )
 endef
 
 define Image/InstallKernel
-	$(call Image/InstallKernel/Template/$(PROFILE))
+	$(foreach profile,$(if $(SUBPROFILES_$(PROFILE)),$(SUBPROFILES_$(PROFILE)),$(PROFILE)),\
+		$(call Image/InstallKernel/Template/$(profile)) )
 endef
 
+
 define Image/Build
-	$(call Image/Build/Profile/$(PROFILE),$(1))
+	$(foreach profile,$(if $(SUBPROFILES_$(PROFILE)),$(SUBPROFILES_$(PROFILE)),$(PROFILE)),\
+		$(call Image/Build/Profile/$(profile),$(1)) )
 endef
 
 $(eval $(call BuildImage))
